tools/data-handler/src/containers/project.ts

```diff
@@ -233,6 +233,34 @@ export class Project extends CardContainer {
     return resources;
   }
 
+  /**
+   * Returns a new unique card key with project prefix (e.g. test_x649it4x).
+   * @returns a new card-key string
+   */
+  public async newCardKey(): Promise<string> {
+    // Create a key and check that there are no collisions with other keys in project
+    // Attempt max 10 times
+    for (let i = 0; i < 10; i++) {
+      const newKey = `${this.settings.cardkeyPrefix}_${this.generateRandomBase36String()}`;
+      const exists = await this.findSpecificCard(newKey);
+      if (exists) continue;
+      return newKey;
+    }
+
+    throw 'Could not generate unique card key';
+  }
+
+  /**
+   * Returns a 8-character random base-36 string (using 0-9 and a-z, e.g. x649it4x)
+   * @returns a random base-36 string
+   */
+  private generateRandomBase36String(): string {
+    // Generate a random number between 0 and 36^8 - 1
```

Comment:
Make the 36, 8 and '0' constants. Then remove the comments.

Reply:
If you move this to utils, it could be a static.
If it would take in a parameter instead of 36 (rename it then), it could be used in the future (big maybe) for something else as well.

Reply:
Changed in to an utils function and added test in [852fc33](https://github.com/CyberismoCom/cyberismo/pull/127/commits/852fc3324963ad17dac30dd7baa1091d6b0f5c39)